mteb/abstasks/mteb_result.py

```diff
@@ -0,0 +1,182 @@
+    """A class to represent the MTEB result.
+
+    Attributes:
+        task_name: The name of the MTEB task.
+        dataset_revision: The revision dataset for the task on HuggingFace dataset hub.
+        mteb_version: The version of the MTEB used to evaluate the model.
+        scores: The scores of the model on the dataset. The scores is a dictionary with the following structure; dict[Split, dict[HFSubset, Scores]].
+            Where Scores is a dictionary with the following structure; dict[str, Any]. Where the keys and values are scores. Split is the split of
+            the dataset. HFSubset is the subset name on huggingface e.g. "en-de". HFSubset is set to "" if there is no subsets on huggingface.
+
+    Example:
+        >>> scores = {
+        ...     "train": {
+        ...         "en-de": {
+        ...             "main_score": 0.5,
+        ...             "evaluation_time": 100,
+        ...         },
+        ...         "en-fr": {
+        ...             "main_score": 0.6,
+        ...             "evaluation_time": 200,
+        ...         },
+        ...     },
+        ... }
+        >>> mteb_results = MTEBResults(
+        ...     dataset_revision="1.0",
+        ...     task_name="sample_task",
+        ...     mteb_version="1.0",
+        ...     scores=scores,
+        ... )
+        >>> mteb_results.get_main_score()  # get the main score for all languages
+        0.55
+        >>> mteb_results.get_main_score(languages=["fra"])  # get the main score for French
+        0.6
+    """
+
+    dataset_revision: str
+    task_name: str
+    mteb_version: str
+    scores: dict[Split, dict[HFSubset, Scores]]
+
+    @field_validator("scores")
+    def _validate_scores(
+        cls, v: dict[Split, dict[HFSubset, Scores]]
+    ) -> dict[Split, dict[HFSubset, Scores]]:
+        for split, hf_subset_scores in v.items():
+            for lang, scores in hf_subset_scores.items():
+                cls._validate_scores_dict(scores)
+        return v
+
+    @staticmethod
+    def _validate_scores_dict(scores: Scores) -> None:
+        if "evaluation_time" not in scores:
+            raise ValueError("'evaluation_time' should be in scores")
+
+        if "main_score" not in scores:
+            raise ValueError("'main_score' should be in scores")
+
+        # check that it is json serializable
+        try:
+            _ = json.dumps(scores)
+        except Exception as e:
+            raise ValueError(f"Scores are not json serializable: {e}")
+
+    def to_dict(self) -> dict:
+        return self.model_dump()
+
+    @classmethod
+    def from_dict(cls, data: dict) -> MTEBResults:
+        return cls.model_validate(data)
+
+    def to_disk(self, path: Path) -> None:
+        with path.open("w") as f:
+            f.write(self.model_dump_json())
+
+    @classmethod
+    def from_disk(cls, path: Path) -> MTEBResults:
+        with path.open("r") as f:
+            return cls.model_validate(json.load(f))
+
+    def get_score(
```

Comment:
If this is not intended to be used, could it be merged with `get_main_score`?

Reply:
This is the generalized case of the get_main_score. My guess is that most will use get main score (as it is simplified), but this function is more flexible allowing you to arbitrarily extract from the scores and I could imagine it being used either internally in the package or by users (e.g. to get evaluation times or get uncertainties). get_main_score is just a wrapper.